tests/apollo/test_skvbc_dbsnapshot.py

```diff
@@ -192,6 +225,68 @@ async def test_db_checkpoint_cleanup(self, bft_network, tracker):
+        """
+        bft_network.start_all_replicas()
+        client = bft_network.random_client()
+        skvbc = kvbc.SimpleKVBCProtocol(bft_network, tracker)
+        for i in range(200):
+            await skvbc.send_write_kv_set()
+
+        op = operator.Operator(bft_network.config, client, bft_network.builddir)
+        rep = await op.create_dbcheckpoint_cmd()
+        data = cmf_msgs.ReconfigurationResponse.deserialize(rep)[0]
+        assert(data.success == True)
+
+        """
+        dbcheckpoint will be created on next stable seq num 300.
+        Adding a 10-second delay to ensure that a dummy client request is sent and reaches a stable seq number of 300.
+        """
+        time.sleep(10)
+
+        getrep = await op.get_dbcheckpoint_info_request()
+        rsi_rep = client.get_rsi_replies()
+        data = cmf_msgs.ReconfigurationResponse.deserialize(getrep)[0]
+        assert(data.success == True)
+        for r in rsi_rep.values():
+            res = cmf_msgs.ReconfigurationResponse.deserialize(r)
+            assert(len(res[0].response.db_checkpoint_info) == 1)
+            dbcheckpoint_info_list = res[0].response.db_checkpoint_info
+            assert(any(dbcheckpoint_info.seq_num == 300 for dbcheckpoint_info in dbcheckpoint_info_list))
+    
+    @with_trio
+    @with_bft_network(start_replica_cmd, selected_configs=lambda n, f, c: n == 7)
+    @verify_linearizability()
+    async def test_get_dbcheckpoint_info_request_cmd(self, bft_network, tracker):
+        """
+            sends a getdbCheckpointInfoRequest command and test for created dbcheckpoints.
+        """
+        bft_network.start_all_replicas()
+        client = bft_network.random_client()
+        skvbc = kvbc.SimpleKVBCProtocol(bft_network, tracker)
+        for i in range(300): 
+            await skvbc.send_write_kv_set()
+        # There will be 2 dbcheckpoints created on stable seq num 150 and 300.
+        await self.wait_for_stable_checkpoint(bft_network, bft_network.all_replicas(), 300)
+        num_of_db_snapshots =  await bft_network.get_metric(0, bft_network, "Counters", "numOfDbCheckpointsCreated", component="rocksdbCheckpoint")
+        assert num_of_db_snapshots == 2
+        
+        op = operator.Operator(bft_network.config, client, bft_network.builddir)
+        rep = await op.get_dbcheckpoint_info_request()
+        rsi_rep = client.get_rsi_replies()
+        data = cmf_msgs.ReconfigurationResponse.deserialize(rep)[0]
+        assert(data.success == True)
+        for r in rsi_rep.values():
+            res = cmf_msgs.ReconfigurationResponse.deserialize(r)
+            assert(len(res[0].response.db_checkpoint_info) == 2)
+            dbcheckpoint_info_list = res[0].response.db_checkpoint_info
+            assert(any(dbcheckpoint_info.seq_num == 150 for dbcheckpoint_info in dbcheckpoint_info_list))
+            assert(any(dbcheckpoint_info.seq_num == 300 for dbcheckpoint_info in dbcheckpoint_info_list))
```

Comment:
same comment as above

Reply:
Fixed. Verify that last dbCheckpoint is created with last block id 300.